service/controller/v13/resource/cloudformation/create.go

```diff
@@ -110,6 +111,18 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
 	if currentStackState.Name == "" || desiredStackState.Name != currentStackState.Name {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster main stack has to be created")
 
+		{
+			r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if network segment has been allocated")
+
+			cidr := key.ClusterNetworkCIDR(customObject)
+			if cidr == "" {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster network segment has not been allocated yet")
```

Comment:
Does this need additional note for skipping current round of execution for this resource's `ApplyCreate()`?